### PR TITLE
Add support for aws_appautoscaling_target

### PR DESCRIPTION
**Pull request by sundowndev**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #977 
| ❓ Documentation  | https://github.com/cloudskiff/driftctl-docs/pull/116 

## Description

```
Found changed resources:
  From tfstate://../tf+sundowndev/terraform.tfstate
    - table/GameScores (aws_appautoscaling_target.dynamodb_table_read_target):
        ~ min_capacity: 5 => 6
    - GameScores (aws_dynamodb_table.basic-dynamodb-table):
        ~ read_capacity: 5 => 6
Found 3 resource(s)
 - 100% coverage
 - 3 resource(s) managed by terraform
     - 2/3 resource(s) out of sync with Terraform state
 - 0 resource(s) not managed by Terraform
 - 0 resource(s) found in a Terraform state but missing on the cloud provider
Scan duration: 10s
Provider version used to scan: 3.47.0. Use --tf-provider-version to use another version.
```

Following this PR, documentation must be updated : 

- Supported resources
- Minimal AWS IAM
  -  **Application Auto Scaling** : DescribeScalableTargets

### CODE REVIEW

File: pkg/remote/aws/repository/appautoscaling_repository.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/aws/aws-sdk-go/service/applicationautoscaling"
+	"github.com/aws/aws-sdk-go/service/applicationautoscaling/applicationautoscalingiface"
+	"github.com/cloudskiff/driftctl/pkg/remote/cache"
+)
+
+type AppAutoScalingRepository interface {
+	ServiceNamespaceValues() []string
+	DescribeScalableTargets(string) ([]*applicationautoscaling.ScalableTarget, error)
+	DescribeScalingPolicies(string) ([]*applicationautoscaling.ScalingPolicy, error)
+	DescribeScheduledActions(string) ([]*applicationautoscaling.ScheduledAction, error)
```
Comment (eliecharra) [inline pkg/remote/aws/repository/appautoscaling_repository.go:16]: Are theses methods useful ? I do not see any usage within enumerator.

**Comment by eliecharra** [comment]
You should update the doc repo too to add this new supported resource to the list

**Comment by eliecharra** [comment]
It misses test for state reader

File: pkg/remote/aws/repository/appautoscaling_repository.go (hunk 1 above)
Comment (sundowndev) [reply pkg/remote/aws/repository/appautoscaling_repository.go:16]: They will be used by #1028  & #1027, as these 3 resources must be implemented together

Do you want me to split accordingly ?

Comment (eliecharra) [reply pkg/remote/aws/repository/appautoscaling_repository.go:16]: Yeah it's better to keep PR atomic 🙏🏻

**Comment by sundowndev** [comment]
> It misses test for state reader

- Added state reader and acceptance tests
- Removed code non-related to `aws_appautoscaling_target`

File: pkg/iac/terraform/state/test/aws_appautoscaling_target/.terraform.lock.hcl [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# This file is maintained automatically by "terraform init".
```
Comment (eliecharra) [inline pkg/iac/terraform/state/test/aws_appautoscaling_target/.terraform.lock.hcl:1]: This file is useless for this test

File: pkg/iac/terraform/state/test/aws_appautoscaling_target/terraform.tf [hunk 3]
```diff
@@ -0,0 +1,61 @@
+provider "aws" {
```
Comment (eliecharra) [inline pkg/iac/terraform/state/test/aws_appautoscaling_target/terraform.tf:1]: This is not very usefull to put the tf file for the IaC test, the state file in json is enough, if needed we can mutate it manually

File: pkg/iac/terraform/state/test/aws_appautoscaling_target/terraform.tfstate.backup [hunk 4]
```diff
@@ -0,0 +1,107 @@
+{
```
Comment (eliecharra) [inline pkg/iac/terraform/state/test/aws_appautoscaling_target/terraform.tfstate.backup:1]: The backup file is not required

**Review by eliecharra — CHANGES_REQUESTED** [review]
Only some useless file cleanup in IaC test folder and we can merge

**Review by eliecharra — APPROVED** [review]

**Pull request merged**